core/math/rect2.h

```diff
@@ -531,11 +531,11 @@ struct _NO_DISCARD_ Rect2i {
 			begin.y = p_vector.y;
 		}
 
-		if (p_vector.x > end.x) {
-			end.x = p_vector.x;
+		if (p_vector.x >= end.x) {
+			end.x = p_vector.x + 1;
```

Comment:
That seems very weird to me. IMO the current behavior was correct, and if the documentation makes it sound like it should be different, then the documentation should be changed.

I would definitely be confused if `Rect2i(0, 0, 10, 10).expand(Vector2i(20, 10))` returns `Rect2i(0, 0, 21, 11)`.